src/events/actions/updateWebsiteActions/generateVoxxrinJson.ts

```diff
@@ -0,0 +1,203 @@
+        end: event.dates.end?.toISOString(),
+        days: days.map((day) => ({
+            id: day.start.toISO() || Math.random().toString(),
+            localDate: day.start.toLocaleString(),
+        })),
+        logoUrl: logoUrl,
+        backgroundUrl: backgroundUrl,
+        theming: {
+            colors: {
+                primaryHex: event.color || '#F78125',
+                primaryContrastHex: hexContrast(event.color || '#F78125'),
+                secondaryHex: event.colorSecondary || '#3880FF',
+                secondaryContrastHex: hexContrast(event.colorSecondary || '#3880FF'),
+                tertiaryHex: event.colorBackground || '#202020',
+                tertiaryContrastHex: hexContrast(event.colorBackground || '#202020'),
+            },
+        },
+        supportedTalkLanguages: supportedTalkLanguages,
+        rooms: event.tracks.map((track) => ({ id: track.id, title: track.name })),
+        talkTracks: event.categories.map((category) => ({
+            id: category.id,
+            title: category.name,
+            themeColor: category.color || '#000000',
+        })),
+        talkFormats: event.formats.map((format) => ({
+            id: format.id,
+            title: format.name,
+            duration: `PT${format.durationMinutes}m`,
+            themeColor: '#000000',
+        })),
+        features: {
+            favoritesEnabled: true,
+            roomsDisplayed: true,
+            remindMeOnceVideosAreAvailableEnabled: false,
+            showInfosTab: true,
+            showRoomCapacityIndicator: false,
+            hideLanguages: [],
+            ratings: {
+                scale: {
+                    enabled: false,
+                },
+                bingo: {
+                    enabled: false,
+                },
+                'free-text': {
+                    enabled: false,
+                },
+            },
+        },
+        sponsors: sponsors.map((sponsorCategory) => ({
+            type: sponsorCategory.name,
+            typeFontColor: event.color || '#000000',
+            typeBackgroundColor: event.colorBackground || '#E5E4E2',
+            sponsorships: sponsorCategory.sponsors
+                .map((sponsor) => ({
+                    name: sponsor.name,
+                    logoUrl: sponsor.logoUrl,
+                    href: sponsor.website || '',
+                }))
+                .filter((sponsor) => !!sponsor.href && sponsor.href.length > 0),
+        })),
+        sessions: {},
+        speakers: {},
+    }
+
+    sessions.forEach((session) => {
+        if (!session.showInFeedback) {
+            return
+        }
+
+        const tags = []
+
+        const categoryName = session.category
+            ? event.categories.find((category) => category.id === session.category)?.name
+            : []
+
+        if (categoryName) {
+            tags.push(categoryName)
+        }
+
+        voxxrinJson.sessions[session.id] = {
```

Comment:
done!

Reply:
@HugoGresse I can see that `abstract`, `categoryId` and `categoryName` can be nullable (typically, on opening/ending keynotes)

I can easily make a fallback for `abstract` by putting an "empty" abstract

But for categories, this is more complicated as I'm not sure that falling back on an (auto-generated) "Keynote" category would be a good thing (this would fit with SunnyTech .. but would it be the same for other events ?.. not sure)

Is there something you could enforce on your side to avoid nullable categories in sessions ?

Reply:
You're right, Keynotes won't have a category, as well as break, lunch, etc. 
I should not try to enforce this on the UI, the goal of OpenPlanner is not to make decision for the organizing team. 
Can you have sessions without categories or format? Or at least no categories? (the format could be added as a restriction for Voxxrin for example).

Reply:
I'm going to auto-generate an `Unknown` track (/category) for these cases

This will render like this :
<img width="300" alt="Cursor_and_Voxxrin" src="https://github.com/HugoGresse/OpenPlanner/assets/603815/35012ad2-1ed8-4217-852e-e7e00fa1a4e7">